### PR TITLE
feat(shadowdom): turn on ShadowDom Emulated Mode by default.

### LINKED ISSUE #526: Switch ShadowDOM strategy to emulated by default
We must also update benchmarks for this change, and check the ToDo App example.
@vsavkin  has a doc that outlines options.

### PR DESCRIPTION
**Pull request by rkirov**
Closes: #526

### CODE REVIEW

**Comment by rkirov** [comment]
With removal of setting `innerHtml` in `clearNodes` implementation, the ShadowDom emulation strategy is no longer significantly slower from native.

tree benchmark (native):
with viewCaching - 27.22+-16
without viewCaching - ~ 106.05

tree benchmark (emulated):
with viewCaching - 27.30+-8%
without viewCaching - ~ 116.50

File: modules/angular2/test/core/application_spec.js [hunk 1]
```diff
@@ -93,8 +109,8 @@ export function main() {
     it('should display hello world', (done) => {
       var injectorPromise = bootstrap(HelloRootCmp, testBindings);
       injectorPromise.then((injector) => {
-        expect(injector.get(appElementToken)
-            .shadowRoot.childNodes[0].nodeValue).toEqual('hello world!');
+        expect(shadowDomTextContent(injector.get(appElementToken))
```
Comment (vsavkin) [inline modules/angular2/test/core/application_spec.js]: `expect(el).toHaveText` should do the trick

File: modules/angular2/src/core/application.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ import {List, ListWrapper} from 'angular2/src/facade/collection';
 import {Promise, PromiseWrapper} from 'angular2/src/facade/async';
 import {VmTurnZone} from 'angular2/src/core/zone/vm_turn_zone';
 import {LifeCycle} from 'angular2/src/core/life_cycle/life_cycle';
-import {ShadowDomStrategy, NativeShadowDomStrategy} from 'angular2/src/core/compiler/shadow_dom_strategy';
+import {ShadowDomStrategy, NativeShadowDomStrategy, EmulatedUnscopedShadowDomStrategy} from 'angular2/src/core/compiler/shadow_dom_strategy';
```
Comment (vicb) [inline modules/angular2/src/core/application.js]: remove Native ?

**Comment by vicb** [comment]
You need to register the annotation on the `Content` directive into the reflector for benchmarks which use a static config.

**Pull request closed without merging**